sktime/transformations/panel/augmenter.py

```diff
@@ -0,0 +1,1217 @@
+        else:
+            raise ValueError("n must be int or float, not " + str(type(n))) + "."
+        # calculate indices
+        if shuffle_and_stratify and without_replacement and y is not None:
+            idx_list = []
+            sss = StratifiedShuffleSplit(
+                n_splits=int(np.floor(n / n_instances)),
+                test_size=0.5,
+                random_state=random_state,
+            )
+            for idx_a, idx_b in sss.split(X, y):
+                idx_list = idx_a.tolist() + idx_b.tolist()
+            sss = StratifiedShuffleSplit(
+                n_splits=1, test_size=np.mod(n, n_instances), random_state=random_state
+            )
+            for _, idx_b in sss.split(y, y):
+                idx_list += idx_b.tolist()
+        else:
+            raise NotImplementedError("Not implemented yet.")
+        # check number of indices
+        if n != len(idx_list):
+            raise ValueError(
+                "The index list must contain n = "
+                + str(n)
+                + "indices, but contains "
+                + str(len(idx_list))
+                + " indices."
+            )
+
+        X_aug = X.iloc[idx_list]
+        # Need to reset_index to pass index.is_monotonic of
+        # check_pdDataFrame_Series() in datatypes/_series/_check.py
+        X_aug.reset_index(inplace=True, drop=True)
+        if y is not None:
+            y_aug = y.iloc[idx_list]
+            y_aug.index = X_aug.index
+            return X_aug, y_aug, idx_list
+        else:
+            return X_aug, idx_list
+
+    def _plot_augmentation_example(self, X, y):
+        """Plot original and augmented instance examples for each variable.
+
+        This is a wrapper function calling the static function
+        plot_augmentation_example() for compatibility reasons.
+        """
+        plot_augmentation_example(self, X, y)
+
+
+# static functions
+def plot_augmentation_example(
+    fitted_transformer,
+    X,
+    y=None,
+    n_instances_per_variable=5,
+):
+    """Plot original and augmented instance examples for each variable.
+
+    Parameters
+    ----------
+    fitted_transformer: fitted transformer
+        A fitted (augmentation) transformer.
+    X: Panel of pd.DataFrame
+        Uni- or multivariate dataset.
+    y: Series or Panel, optional (default=None)
+        Target variable, if y is available and of categorical scale,
+        it will be used to stratify the randomly drawn examples.
+    n_instances_per_variable: int, optional (default = 5)
+        number of time series to draw per variable (row).
+
+    Returns
+    -------
+    matplotlib.figure.Figure: A figure with a [n_variables, 2] subplot-grid.
+    """
+    _check_soft_dependencies("matplotlib")
+    import matplotlib.pyplot as plt
+
+    n_vars = X.shape[1]  # get number of variables of X
+    # pick (stratified regarding categorical y) examples from the original input
+    # data
+    X, y, idx = SeqAugPipeline.draw_random_samples(
```

Comment:
`SeqAugPipeline` feels a bit unnecessary to call here, maybe a call to a standalone `draw_random_samples` would suffice?

Reply:
Matches with your question above. We first planned to make `draw_random_samples` as indispensable part of the pipeline class (i.e. by calling `transform` or  `fit` on the pipeline random samples are augmented) but this does not comply to a lot of rules of sklearn. So short answer: Yes, I will definitely move  `draw_random_samples` out of the pipeline class.

Reply:
Fixed that (see above...)